Objects/weakrefobject.c

```diff
@@ -1090,6 +1090,12 @@ _PyWeakref_ClearWeakRefsExceptCallbacks(PyObject *obj)
     UNLOCK_WEAKREFS(obj);
 }
 
+void
+PyUnstable_Weakref_ClearWeakRefsExceptCallbacks(PyObject *obj)
+{
```

Comment:
I suggest to check if `_PyType_SUPPORTS_WEAKREFS(Py_TYPE(op))` and do nothing (return) if it's not the case.